src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateEnumActionsUtils.java

```diff
@@ -0,0 +1,230 @@
+package com.commercetools.sync.producttypes.utils;
+
+import com.commercetools.sync.commons.exceptions.DuplicateKeyException;
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.EnumValue;
+import io.sphere.sdk.models.WithKey;
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.producttypes.commands.updateactions.RemoveEnumValues;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Map;
+import java.util.List;
+import java.util.Optional;
+import java.util.Collection;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toMap;
+
+public final class ProductTypeUpdateEnumActionsUtils {
+    /**
+     * Given a list of enum values, gets a map where the keys are the enum value key, and the values
+     * are the enum instances.
+     *
+     * @param enumValues the list of enum values.
+     * @param <T>        the enum type of the elements of the list.
+     * @return a map with the enum value key as a key of the map, and the
+     *         enum value as a value of the map.
+     */
+    @Nonnull
+    public static <T extends WithKey> Map<String, T> getEnumValuesKeyMap(@Nonnull final List<T> enumValues) {
+        return enumValues
+            .stream()
+            .collect(toMap(WithKey::getKey, enumValue -> enumValue));
+    }
+
+    /**
+     * Given a list of new {@link EnumValue}s, gets a map where the keys are the enum value key, and the values
+     * are the enum instances.
+     *
+     * @param attributeDefinitionName the attribute definition name whose the enum values belong to.
+     * @param enumValues              the list of enum values.
+     * @param <T>                     the enum type of the elements of the list.
+     * @return a map with the enum value key as a key of the map, and the enum
+     *         value as a value of the map.
+     * @throws DuplicateKeyException in case there are enum values with duplicate keys.
+     */
+    @Nonnull
+    public static <T extends WithKey> Map<String, T> getEnumValuesKeyMapWithKeyValidation(
+        @Nonnull final String attributeDefinitionName,
+        @Nonnull final List<T> enumValues) throws DuplicateKeyException {
```

Comment:
The `throws DuplicateKeyException` here is redundant since it's a runtime exception

Reply:
c3d86f78c7672ae9811d17d51c2cc8fc3d2d5a13